### ISSUE TITLE
Records can't be pickled because of the Metaclass used to create them.

### ISSUE THREAD

**Issue by grybmadsci** [chainreactionmfg/mutablerecords#3]
I've got a mostly-working fix for this, but the loaded record doesn't compare equal to the dumped record because a new type with the same name gets created (obviously).  I'd like to propose a change to the equality check for `RecordClass` so that it more closely matches the equality check of `namedtuple`s.  `namedtuple` instances compare equal if corresponding fields (by index, not name), compare equal, essentially as if they were normal tuples.  I'd like to propose an update to the equality check of `RecordClass` to one of:
- Records are equal iff all required fields compare equal (compared according to index, not name) and all optional fields compare equal, compared according to name (basically dicts mapping optional field name to value compare equal).  This most closely matches `namedtuple`, but does allow you to do weird things.
- Same as previous, but compare all fields by name, rather than comparing required fields positionally.
- Same as previous, but also require that the Record type names match.

Thoughts?  Any of these work for my purposes, I think the last on is a little too strict of a check, but only because of how loose `namedtuple` comparisons are.

**Comment by fahhem** [comment]
namedtuple is trying to be a barely-better-than-tuple, but mutablerecords isn't. I'd like to compare fields by name since their index doesn't make sense for a record. Comparing the name also makes sense, since you don't want a sub-Record with an extra field to compare equal to its base class (as that will make equality non-symmetric). It's possible a sub-Record with the same name gets past this, but unless we can easily access the module path as well, I'm fine not checking that.

In summary, I'd prefer #3 since the syntax is like `namedtuple` but the semantics aren't entirely

**Comment by grybmadsci** [comment]
SGTM, I'll do it that way.  I agree that the semantics of `namedtuple` are moderately different.  I'm not convinced comparing the names really buys you anything, though, besides perhaps a false sense of security.  If you do an equality check on all attributes, then sub-Record comparisons are symmetrically equal, and the name is arbitrarily set by the user anyway.

For example, you could make two `Record` types that have the same name and same fields but provide different functionality and then they'd compare equal anyway even with the most strict of these checks.

I'll go ahead with the third approach, just throwing some thoughts out there for consideration :)

**Comment by fahhem** [comment]
I'm mostly thinking about the cases where bad equality checks have bit me, such as in dicts. If you have a class that happens to hash the same as another (modulo the dict size), and they compare as equal in Python, but they're not actually equal, you get some weird behavior. I spent hours tracking down why a Record was going into a dict but never coming out without raising an errors, and it was because my equality comparison was too loose.

Oh, I think it's a much simpler fix for you than you might think. If you change line 78 in records.py to say "type(self) == type(other)" then you'll get the metaclass's `__eq__()` invoked, which checks that the attribute names are the same. Then the rest of `RecordClass.__eq__()` checks for the values as you expect (by name).

**Comment by grybmadsci** [comment]
Fixed by #4
